Tests are failing. I think you forget to adapt some test code.
```
server/room_test.go:13:29: not enough arguments in call to r.CreateTable
	have (string, nil, nil)
	want (string, *string, *string, int64)
server/room_test.go:25:26: cannot use 123 (type int) as type *rand.Rand in argument to game.DealCards
server/table_test.go:10:19: not enough arguments in call to NewTable
	have (string, nil, nil)
	want (string, *string, *string, int64)
```

--- a/doko/game/cards.go
+++ b/doko/game/cards.go
@@ -2,10 +2,7 @@ package game
 
 import (
 	"fmt"
-	"log"
 	"math/rand"
-	"os"
-	"strconv"
 	"strings"
 )
 
@@ -87,23 +84,9 @@ func CreateDeck() []Card {
 	return ans
 }
 
-func DealCards(seed int64) Cards {
-	if seed == 0 {
-		seed = rand.Int63()
-	}
-
-	override, isOverwritten := os.LookupEnv("KC_DBG_SEED")
-	if isOverwritten {
-		override, importError := strconv.ParseInt(override, 10, 64)
-		if importError == nil {
-			seed = override
-		}
-	}
-	log.Printf("Dealing cards (seed: %d)", seed)
-
+func DealCards(rng *rand.Rand) Cards {
 	deck := CreateDeck()
-	random := rand.New(rand.NewSource(seed))
-	random.Shuffle(len(deck), func(i int, j int) { deck[i], deck[j] = deck[j], deck[i] })
+	rng.Shuffle(len(deck), func(i int, j int) { deck[i], deck[j] = deck[j], deck[i] })
 	var ans [NumPlayers]Hand
 	for i := 0; i < NumPlayers; i++ {
 		ans[i] = deck[i*NumHandCards : (i+1)*NumHandCards]

--- a/doko/game/cards_test.go
+++ b/doko/game/cards_test.go
@@ -2,6 +2,7 @@ package game
 
 import (
 	"github.com/stretchr/testify/assert"
+	"math/rand"
 	"sort"
 	"testing"
 )
@@ -22,7 +23,8 @@ func TestHand_RemoveCard(t *testing.T) {
 
 //noinspection GoNilness
 func TestDealCards(t *testing.T) {
-	hands := DealCards(126795)
+	rng := rand.New(rand.NewSource(126795))
+	hands := DealCards(rng)
 	var dealtDeck []Card
 	for _, p := range Players() {
 		dealtDeck = append(dealtDeck, hands[p]...)

--- a/doko/game/game_test.go
+++ b/doko/game/game_test.go
@@ -2,6 +2,7 @@ package game
 
 import (
 	"github.com/stretchr/testify/assert"
+	"math/rand"
 	"testing"
 )
 
@@ -13,7 +14,8 @@ func TestWinnerOfTrick(t *testing.T) {
 }
 
 func TestGame_IsValidMove(t *testing.T) {
-	cards := DealCards(1487)
+	rng := rand.New(rand.NewSource(1487))
+	cards := DealCards(rng)
 	game := NewGame(cards, Player3, someNormalspiel)
 	for _, card := range game.HandCards[Player3] {
 		assert.Equal(t, game.CanPlayCard(Player3, card), CardPlayed)

--- a/doko/match/round.go
+++ b/doko/match/round.go
@@ -2,6 +2,7 @@ package match
 
 import (
 	"github.com/supermihi/karlchencloud/doko/game"
+	"math/rand"
 )
 
 const MaxPlayersPerRound = 6
@@ -33,11 +34,11 @@ func NewMatchStats(evaluation *GameEvaluation, players *PlayerAssignment) MatchS
 type Round struct {
 	numPlayers int
 	scores     []MatchStats
-	cardSeed   int64
+	rng        *rand.Rand
 }
 
-func NewRound(numPlayers int, cardSeed int64) *Round {
-	return &Round{numPlayers, make([]MatchStats, 0), cardSeed}
+func NewRound(numPlayers int, rng *rand.Rand) *Round {
+	return &Round{numPlayers, make([]MatchStats, 0), rng}
 }
 func (r *Round) CurrentPlayerAssignment() PlayerAssignment {
 	return NewPlayerAssignment(r.numPlayers, r.NumFinishedGames())
@@ -46,7 +47,7 @@ func (r *Round) CurrentPlayerAssignment() PlayerAssignment {
 func (r *Round) NextMatch() Match {
 	index := r.NumFinishedGames()
 	assignment := NewPlayerAssignment(r.numPlayers, index)
-	return NewMatch(assignment.Forehand(r.numPlayers), game.DealCards(r.cardSeed+int64(index)))
+	return NewMatch(assignment.Forehand(r.numPlayers), game.DealCards(r.rng))
 }
 
 func (r *Round) NumFinishedGames() int {

--- a/server/core.go
+++ b/server/core.go
@@ -68,8 +68,14 @@ func GetData(t *Table) *TableData {
 	if t.playersInOrder == nil {
 		players = t.players
 	}
-	return &TableData{Id: t.Id, Owner: t.Owner(), InviteCode: t.InviteCode, Players: players,
-		Phase: t.Phase, Created: t.Created}
+	return &TableData{
+		Id: t.Id,
+		Owner: t.Owner(),
+		InviteCode: t.InviteCode,
+		Players: players,
+		Phase: t.Phase,
+		Created: t.Created,
+	}
 }
 
 func (d *TableData) ContainsPlayer(id string) bool {

--- a/server/core_test.go
+++ b/server/core_test.go
@@ -4,14 +4,16 @@ import (
 	"github.com/stretchr/testify/assert"
 	"github.com/supermihi/karlchencloud/api"
 	"github.com/supermihi/karlchencloud/doko/match"
+	"math/rand"
 	"testing"
 	"time"
 )
 
 func TestGetData_PlayersAreInGameOrder(t *testing.T) {
+	rng := rand.New(rand.NewSource(123))
 	table := &Table{"123", time.Now(), "123", api.TablePhase_NOT_STARTED,
 		[]string{"p1", "p2", "p3", "p4"}, []string{"p3", "p2", "p4", "p1"},
-		match.NewRound(4, 123), nil}
+		match.NewRound(4, rng), nil, rng}
 	data := GetData(table)
 	assert.Equal(t, []string{"p3", "p2", "p4", "p1"}, data.Players)
 }

--- a/server/room.go
+++ b/server/room.go
@@ -14,11 +14,11 @@ func NewRoom(users Users) *Room {
 	return &Room{users, make(map[string]*Table)}
 }
 
-func (r *Room) CreateTable(owner string, fixedTableId *string, fixedInviteCode *string) (table *TableData, err error) {
+func (r *Room) CreateTable(owner string, fixedTableId *string, fixedInviteCode *string, seed int64) (table *TableData, err error) {
 	if r.ActiveTableOf(owner) != nil {
 		return nil, NewCloudError(TableAlreadyExists)
 	}
-	t := NewTable(owner, fixedTableId, fixedInviteCode)
+	t := NewTable(owner, fixedTableId, fixedInviteCode, seed)
 	r.tables[t.Id] = t
 	return GetData(t), nil
 }

--- a/server/room_test.go
+++ b/server/room_test.go
@@ -4,13 +4,14 @@ import (
 	"github.com/stretchr/testify/assert"
 	"github.com/supermihi/karlchencloud/doko/game"
 	"github.com/supermihi/karlchencloud/doko/match"
+	"math/rand"
 	"testing"
 )
 
 func TestRoom_JoinTable(t *testing.T) {
 	r := NewRoom(NewMemoryUserDb())
 	assert.True(t, r.users.Add("owner", "owner", "secret"))
-	table, err := r.CreateTable("owner", nil, nil)
+	table, err := r.CreateTable("owner", nil, nil, 0)
 	assert.Nil(t, err)
 	_, err = r.JoinTable("player 2", "not the invite code")
 	assert.NotNil(t, err)
@@ -22,7 +23,8 @@ func TestRoom_JoinTable(t *testing.T) {
 
 
 func TestRoom_GetMatchData(t *testing.T) {
-	cards := game.DealCards(123)
+	rng := rand.New(rand.NewSource(123))
+	cards := game.DealCards(rng)
 	match := match.NewMatch(game.Player1, cards)
 	for _, player := range game.PlayersFrom(game.Player1) {
 		match.AnnounceGameType(player, game.NormalspielType)

--- a/server/server.go
+++ b/server/server.go
@@ -49,7 +49,13 @@ func (s *dokoserver) CreateTable(ctx context.Context, _ *api.Empty) (*api.TableD
 	user, _ := GetAuthenticatedUser(ctx)
 	s.roomMtx.Lock()
 	defer s.roomMtx.Unlock()
-	table, err := s.room.CreateTable(user.Id, &s.config.Room.ConstantTableId, &s.config.Room.ConstantInviteCode)
+
+	table, err := s.room.CreateTable(
+		user.Id,
+		&s.config.Room.ConstantTableId,
+		&s.config.Room.ConstantInviteCode,
+		s.config.Room.Seed(),
+	)
 	if err != nil {
 		return nil, toGrpcError(err)
 	}

--- a/server/table.go
+++ b/server/table.go
@@ -5,6 +5,7 @@ import (
 	"github.com/supermihi/karlchencloud/api"
 	"github.com/supermihi/karlchencloud/doko/game"
 	"github.com/supermihi/karlchencloud/doko/match"
+	"log"
 	"math/rand"
 	"time"
 )
@@ -18,6 +19,7 @@ type Table struct {
 	playersInOrder []string
 	round          *match.Round
 	CurrentMatch   *TableMatch
+	Rng            *rand.Rand
 }
 
 func (t *Table) Owner() string {
@@ -28,7 +30,8 @@ func (t *Table) String() string {
 	return fmt.Sprintf("Table %v", t.Id)
 }
 
-func NewTable(owner string, fixedTableId *string, fixedInviteCode *string) *Table {
+func NewTable(owner string, fixedTableId *string, fixedInviteCode *string, seed int64) *Table {
+	log.Printf("Creating table (seed: %d)", seed)
 	return &Table{
 		getStringWithDefault(fixedTableId, randomTableId),
 		time.Now(),
@@ -38,6 +41,7 @@ func NewTable(owner string, fixedTableId *string, fixedInviteCode *string) *Tabl
 		nil,
 		nil,
 		nil,
+		rand.New(rand.NewSource(seed)),
 	}
 }
 
@@ -58,10 +62,10 @@ func (t *Table) Start() error {
 	}
 	t.playersInOrder = make([]string, len(t.players))
 	copy(t.playersInOrder, t.players)
-	rand.Shuffle(len(t.players), func(i int, j int) {
+	t.Rng.Shuffle(len(t.players), func(i int, j int) {
 		t.playersInOrder[i], t.playersInOrder[j] = t.playersInOrder[j], t.playersInOrder[i]
 	})
-	t.round = match.NewRound(len(t.players), rand.Int63())
+	t.round = match.NewRound(len(t.players), t.Rng)
 	t.Phase = api.TablePhase_BETWEEN_GAMES
 	return t.StartMatch()
 }

--- a/server/table_test.go
+++ b/server/table_test.go
@@ -7,7 +7,7 @@ import (
 )
 
 func TestTable_Start(t *testing.T) {
-	table := NewTable("user1", nil, nil)
+	table := NewTable("user1", nil, nil, 0)
 	_ = table.Join("user2")
 	_ = table.Join("user3")
 	_ = table.Join("user4")

